library/firewall_lib.py

```diff
@@ -1140,6 +1213,85 @@ def exception_handler(exception_message):
                 else:
                     module.fail_json(msg="INVALID SERVICE - " + item)
 
+    # ipset operations
+    if ipset_operation:
+        ipset_exists = ipset in fw.config().getIPSetNames()
+        fw_ipset = None
+        fw_ipset_settings = None
+        if ipset_exists:
+            fw_ipset = fw.config().getIPSetByName(ipset)
+            fw_ipset_settings = fw_ipset.getSettings()
+            if ipset_type and ipset_type != fw_ipset_settings.getType():
+                module.fail_json(
+                    msg="Name conflict when creating ipset - "
+                    "ipset %s of type %s already exists"
+                    % (ipset, fw_ipset_settings.getType())
+                )
+        elif state == "present":
+            fw_ipset, fw_ipset_settings = create_ipset(module, fw, ipset, ipset_type)
+            changed = True
+            ipset_exists = True
+        if state == "present":
+            if (
+                description is not None
+                and description != fw_ipset_settings.getDescription()
+            ):
+                if not module.check_mode:
+                    fw_ipset_settings.setDescription(description)
+                changed = True
+            if short is not None and short != fw_ipset_settings.getShort():
+                if not module.check_mode:
+                    fw_ipset_settings.setShort(short)
+                changed = True
+            for entry in ipset_entries:
```

Comment:
what if the list of `ipset_entries` is smaller than the existing list of ipset entries for this ipset?  should the code remove entries that aren't in the given `ipset_entries`?

Reply:
with `state: present`, `ipset_entries` will add the entry if it is not in the ipset, and do nothing otherwise. If 
`state: absent`, `ipset_entries` will remove the entries if they are present in the ipset, and do nothing if they are not present.

Would it be better if we simply had `ipset_entries` replace the entries in an ipset when specified with `state: present`, then not work with `state: absent`?

Reply:
> with `state: present`, `ipset_entries` will add the entry if it is not in the ipset, and do nothing otherwise. If `state: absent`, `ipset_entries` will remove the entries if they are present in the ipset, and do nothing if they are not present.

It should be consistent with how other types work - so if services work like this, then ipset should also.

> 
> Would it be better if we simply had `ipset_entries` replace the entries in an ipset when specified with `state: present`, then not work with `state: absent`? This was one design that I had considered, but decided against due in favor of consistency with how services are defined, and how the firewall ansible_fact allows the user to see the full configuration of the ipset.